### ISSUE TITLE
JVM: isAny() (and other methods) doesn't work on interfaces

### ISSUE THREAD

**Issue by glureau** [Kodein-Framework/MocKMP#7]
```kotlin
interface Bar
class Foo { fun consume(bar: Bar) }
  // in a test
  mocker.verify { foo.consume(isAny<Bar>()) } // throws a java.lang.InstantiationError
```

Objenesis can instantiate classes but I think it cannot instantiate interface as far as I can tell:

```kotlin
    class Foo
    interface Bar
    val cl = ObjenesisStd(true).newInstance(Foo::class.java) // Builds a Foo
    val inter = ObjenesisStd(true).newInstance(Bar::class.java) // throws InstantiationError
```

I suppose Mockmp should detect if it's an interface and may:
- if the interface has a generated Mock, use the Mock class instead
- if no mock has been generated for this interface, then provides an error message explaining the limitation and procedure to fix it.

**Comment by SalomonBrys** [comment]
Looks like ObjGenesis cannot instanciate Arrays as well: https://kotlinlang.slack.com/archives/D035VNK0GTF/p1646669455939899

**Comment by SalomonBrys** [comment]
Fixed in 1.3.0.

https://github.com/Kodein-Framework/MocKMP/releases/tag/v1.3.0
